keyboards/planck/keymaps/juno/keymap.c

```diff
@@ -0,0 +1,458 @@
+	_______, KC_MPRV, KC_MPLY, KC_MNXT, BACKLIT, KC_END,  _______, _______, KC_VOLD, KC_VOLU, KC_MUTE, _______,
+	KC_CAPS, _______, _______, _______, _______, _______, _______, _______, DP_ON,   _______, _______, _______
+),
+
+/* DPAD layer, to activate arrow cluster at the bottom right corner
+
+ * ,-----------------------------------------------------------------------------------.
+ * | ESC  |   Q  |   W  |   E  |   R  |   T  |   Y  |   U  |   I  |   O  |   P  | Bksp |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * | Tab  |   A  |   S  |   D  |   F  |   G  |   H  |   J  |   K  |   L  |   ;  |  "   |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * | Shift|   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |  Up  |Enter |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | Ctrl | GUI  | Alt  |      |DNUM  |    Space    |      |DP_OFF| Left | Down |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+
+[_DPAD] = LAYOUT_planck_grid(
+	KC_ESC,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSPC,
+	KC_TAB,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
+	KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_UP,   KC_ENT ,
+	KC_LCTL, KC_LGUI, KC_LALT, XXXXXXX, MO(_DPADNUM),
+												 KC_SPC,  KC_SPC,  LOWER,   DP_OFF,  KC_LEFT, KC_DOWN, KC_RGHT
+),
+
+// Extended layer for weapon switching
+
+[_DPADNUM] = LAYOUT_planck_grid(
+	XXXXXXX, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    XXXXXXX,
+	XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,
+	_______, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
+	_______, _______, _______, XXXXXXX, MO(_DPADNUM),
+												 XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX
+)
+
+
+};
+
+#ifdef AUDIO_ENABLE
+  float plover_song[][2]     = SONG(PLOVER_SOUND);
+  float plover_gb_song[][2]  = SONG(PLOVER_GOODBYE_SOUND);
+  
+  // Borrowing audio from unused audio
+  // Caps Lock on and off sound too similar
+  
+  float caps_song_on[][2] = SONG(NUM_LOCK_ON_SOUND);
+  float caps_song_off[][2] = SONG(SCROLL_LOCK_ON_SOUND);
+
+  float dpad_song_on[][2] = SONG(ZELDA_PUZZLE);
+  float dpad_song_off[][2] = SONG(SONIC_RING);
+#endif
+
+uint32_t layer_state_set_user(uint32_t state) {
+
+  // LED control, lighting up when Fn layer is activated
+  switch (biton32(state)) {
+    case _QWERTY:
+      backlight_set(0);
+      break;
+    case _FN1:
+      backlight_set(3);
+      break;
+	case _DPAD:
+      backlight_set(3);
+      break;
+  }
+
+  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
```

Comment:
It' better if this is called first.  That way, you can do stuff based on the new state.  But for what you have here, this is fine.

Reply:
Not sure if I misunderstood anything, but this:

```
uint32_t layer_state_set_user(uint32_t state) {
	
  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);

  // LED control, lighting up when Fn layer is activated
  
  switch (biton32(state)) {
    case _QWERTY:
      backlight_set(0);
      break;
    case _FN1:
      backlight_set(3);
      break;
	case _DPAD:
      backlight_set(3);
      break;
  }
}
```

apparently doesn't work, as the ```return``` would terminate the function.

Reply:
Specifically: 
```c
int32_t layer_state_set_user(uint32_t state) {
	
  state = update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);

  // LED control, lighting up when Fn layer is activated
  
  switch (biton32(state)) {
    case _QWERTY:
      backlight_set(0);
      break;
    case _FN1:
      backlight_set(3);
      break;
	case _DPAD:
      backlight_set(3);
      break;
  }
  return state;
}

Reply:
Right! My poor understanding of the API and unwillingness to tinker. Thanks for the help.

This would be updated in one of the upcoming  commits.